mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -94,10 +106,59 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
       }
     }
 
+    // Obtain convolution parameters: padding / dialtion / stride.
+    auto leftPadH =
+        paddingAttr.getValue()[0].template dyn_cast<IntegerAttr>().getInt();
+    auto leftPadW =
+        paddingAttr.getValue()[1].template dyn_cast<IntegerAttr>().getInt();
+    auto dilationH =
+        dilationsAttr.getValue()[0].template dyn_cast<IntegerAttr>().getInt();
+    auto dilationW =
+        dilationsAttr.getValue()[1].template dyn_cast<IntegerAttr>().getInt();
+    auto strideH =
+        stridesAttr.getValue()[0].template dyn_cast<IntegerAttr>().getInt();
+    auto strideW =
+        stridesAttr.getValue()[1].template dyn_cast<IntegerAttr>().getInt();
+
+    // get y, x, ho, wo, hi, wi
+    int64_t y, x, ho, wo, hi, wi;
+    y = x = ho = wo = hi = wi = 0;
+    for (unsigned i = 0; i < 4; ++i) {
+      auto filterAttr =
+          filterLayoutAttr.getValue()[i].template dyn_cast<StringAttr>();
+      auto inputAttr =
+          inputLayoutAttr.getValue()[i].template dyn_cast<StringAttr>();
+      auto outputAttr =
+          outputLayoutAttr.getValue()[i].template dyn_cast<StringAttr>();
+
+      if (filterAttr.getValue() == "y") {
+        y = filterShape[i];
+      } else if (filterAttr.getValue() == "x") {
+        x = filterShape[i];
+      }
+
+      if (inputAttr.getValue() == "hi") {
+        hi = inputShape[i];
+      } else if (inputAttr.getValue() == "wi") {
+        wi = inputShape[i];
+      }
+
+      if (outputAttr.getValue() == "ho") {
+        ho = outputShape[i];
+      } else if (outputAttr.getValue() == "wo") {
+        wo = outputShape[i];
+      }
+    }
```

Comment:
I feel like it is easier to use process the dimension/index values into a StringMap. In this case, you probably only need dimension as the value.

https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/blob/31d92f4c64ae6fa6b7c5d543f68b69300b4513ce/mlir/include/mlir/Dialect/MIOpen/Tuning/ConvContext.h#L150-L154

Reply:
@jerryyin Refactoring variable names is better handled in a subsequent PR. Could you help create a ticket for that?